extensions/mongodb-client/deployment/pom.xml

```diff
@@ -58,6 +58,11 @@
             <artifactId>assertj-core</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-smallrye-metrics-deployment</artifactId>
```

Comment:
This should be `quarkus-smallrye-metrics-spi`

Reply:
quarkus-smallrye-metrics-spi does not have the MetricRegistries class.
If I use an `@Inject RegistryType(VENDOR) MetricRegistry` for the Metric Registry, arc can not find an implementation.

How can I just use quarkus-smallrye-metrics-spi in my tests?

Reply:
`quarkus-smallrye-metrics-spi` basically only contains the `MetricBuildItem` class which is used for registering metrics at build time. If you need your tests to use the metrics extension, importing `quarkus-smallrye-metrics-deployment` is fine IMO. 
Instead of using `MetricRegistries`, which is a class from SmallRye Metrics, I'd suggest staying with just pure MP Metrics API, that is indeed `@Inject RegistryType(VENDOR) MetricRegistry`. If you import `quarkus-smallrye-metrics-deployment` it should work I think

Reply:
> If you need your tests to actually use the metrics extension, importing quarkus-smallrye-metrics-deployment is fine IMO.

I'm not agree, if your test need `quarkus-smallrye-metrics-deployment` you should add it with the `test` scope and also import the SPI. With this, someone that don't want metrics can still avoid the inclusion of all the dependencies at runtime.

Reply:
- it has the `test` scope here
- What's the point in importing the SPI when you're not using anything that it offers?

Reply:
@jmartisk sorry, I didn't notice that it has the test scope.

The fact that it didn't use the `MetricBuildItem` is a little annoying as it will be better than all extension register metrics the same way.

Reply:
I agree it would be convenient, and we try to do that as much as possible, but it's not achievable always. Not always do extensions know the full list of metrics at build time.